src/components/MAPI/apps/ClusterDetailWidgetApps.tsx

```diff
@@ -74,6 +76,30 @@ const ClusterDetailWidgetApps: React.FC<IClusterDetailWidgetAppsProps> = (
   const hasNoApps =
     typeof appCounters.apps === 'number' && appCounters.apps === 0;
 
+  const upgradableAppsKey = appList
+    ? getUpgradableAppsKey(appList.items)
+    : null;
+
+  const { data: upgradableApps, error: upgradableAppsError } = useSWR<
+    string[],
+    GenericResponseError
+  >(upgradableAppsKey, () =>
+    getUpgradableApps(appList!.items, clientFactory, auth)
```

Comment:
This works well, but I have a concern, because we're fetching so many `AppCatalogEntry` CRs, that we're also fetching in other parts of the app, and we're not making use of the caching in `swr`. Is there any way that we can leverage the cache?